### PR TITLE
Feature#20/generate employees reports according their attendance

### LINKED ISSUE #20: Generate employees reports according their attendance
# REQUIREMENTS
Display in the view the attendance by day.
Display the average time employees checked in/out by month.
Display the absence reports by months.

# DESIGN
Not necessary.

### PR DESCRIPTION
**Pull request by Frcomparan**
# Description
What did you implemented/Why did you implemented this:
What?
 - Create "absence by month" report
   - Show how many absences have all employees in the current month (or any month you select)
   - Show how many absences have the employee you select (using the ID)
 - Create "attendance by day" report
   - Show which employees have an attendance in the current day (or any day you select)
   - Create a filter to search the attendance of one employee (typing her/his id)
   - Create "average time employees checked in/out by month" report
   - Show the average time of checked in/out of all employee in the current month (or any month you select)  
 
Why?
 - To generate friendly view to show all necessary  reports for admins

  ## Screenshots

image

### CODE REVIEW

File: app/assets/stylesheets/sass/views/attendances/_index.css.scss [hunk 1]
```diff
@@ -126,4 +133,43 @@
...
+  position:relative;
+}
+.table-scroll {
+  max-height:60rem;
+  overflow:auto;
+}
+.table-wrapper table {
+  width:100%;
+}
+
```
Comment (dannharo) [inline app/assets/stylesheets/sass/views/attendances/_index.css.scss]: left an empty line between classes. also a space after the `colon` 

```
position: relative;
```
not 
```
position:relative;
```
check the whole file

File: app/controllers/attendances_controller.rb [hunk 2]
```diff
@@ -16,15 +19,29 @@ def new
...
 
   # POST /attendances or /attendances.json
   def create
     @employee = Employee.find_by(private_code: attendance_params[:employee_id])
-    if @employee && !helpers.check_complete? then new_check_in
-    elsif @employee && helpers.check_complete? then new_check_out
-    else redirect_to new_attendance_path, alert: "Employee doesn't exist"
+    return redirect_to new_attendance_path, alert: "Employee doesn't exist" unless @employee
+    
+    @attendance = @employee.attendances.where(check_in: DateTime.now.beginning_of_day..DateTime.now.end_of_day).last
```
Comment (dannharo) [inline app/controllers/attendances_controller.rb]: you could add a rails [scope](https://guides.rubyonrails.org/active_record_querying.html#scopes)` in the `attendance` model to make it clear here, wdt?

**Review by dannharo — COMMENTED** [review]
good job, just some suggestions

File: app/assets/stylesheets/sass/views/attendances/_index.css.scss (hunk 1 above)
Comment (Frcomparan) [reply app/assets/stylesheets/sass/views/attendances/_index.css.scss]: Change made, file was fixed

File: app/controllers/attendances_controller.rb (hunk 2 above)
Comment (Frcomparan) [reply app/controllers/attendances_controller.rb]: @dannharo, change made, the scope was added

**Review by YilKanda — APPROVED** [review]
LGTM!

**Review by MrCano369x — APPROVED** [review]
LGTM!

**Review by AlejandroCuevasZepeda — APPROVED** [review]
LGTM

**Review by dannharo — COMMENTED** [review]
[Remember that you have a non-functional requirement to add specs on your code](https://github.com/BC-MAY-21-ROR/kata-07-reloj-checador-k7-f22-t4#requerimientos-no-funcionales)

**Pull request merged**